src/authentication/background/auth-firebase.ts

```diff
@@ -0,0 +1,68 @@
+import {
+    AuthenticatedUser,
+    AuthenticatedUserWithClaims,
+    AuthEvents,
+    AuthInterface,
+} from 'src/authentication/background/types'
+import { firebase } from 'src/util/firebase-app-initialized'
+import 'firebase/functions'
+import 'firebase/auth'
+import { FirebaseFunctionsAuth } from 'src/authentication/background/firebase-functions-subscription'
+import { RemoteEventEmitter } from 'src/util/webextensionRPC'
+import { Claims } from 'firebase-backend/firebase/functions/src/types'
+
+export class AuthFirebase implements AuthInterface {
+    private firebaseAuthObserver: firebase.Unsubscribe
+
+    private firebaseAuthFunctions = new FirebaseFunctionsAuth()
+    private authEmitter: RemoteEventEmitter<AuthEvents>
+
+    private getUserFromFirebaseUser(user): AuthenticatedUser | null {
+        if (user == null) {
+            return null
+        }
+        return {
+            uid: user.uid,
+            displayName: user.displayName,
+            email: user.email,
+            emailVerified: user.emailVerified,
+        }
+    }
+
+    async getCurrentUser(): Promise<AuthenticatedUserWithClaims | null> {
+        const user = firebase.auth().currentUser
+        return this.withClaims(this.getUserFromFirebaseUser(user))
+    }
+
+    registerAuthEmitter = authEmitter => {
```

Comment:
Maybe it'd be better to have a public `EventEmitter` on this class, and use that from the outside to send events across the IPC border?